src/main.rs

```diff
@@ -0,0 +1,61 @@
+fn main() {
+    println!("Hello, world!");
+    let (tx, rx) = mpsc::channel();
+    let kura = Kura { tx };
+    thread::spawn(move || loop {
+        kura.tx.send("Test".to_string()).unwrap();
+        thread::sleep(Duration::from_millis(200));
+    });
+    let wsv = Wsv {
+        inner: Arc::new(Mutex::new(Vec::new())),
+        rx: Arc::new(Mutex::new(rx)),
+    };
+    let mut torii = Torii { wsv };
+    torii.start();
+}
+
+use std::{
+    sync::{mpsc, Arc, Mutex},
+    thread,
+    time::Duration,
+};
+
+struct Torii {
+    wsv: Wsv,
+}
+
+impl Torii {
+    pub fn start(&mut self) {
+        self.wsv.start();
+        loop {
+            self.wsv.read();
+            thread::sleep(Duration::from_millis(100));
+        }
+    }
+}
+
+struct Wsv {
```

Comment:
should probably add some comments so random NPCs who look at this code can figure out why it is here